src/appleseed/foundation/image/color.h

```diff
@@ -40,7 +40,14 @@
 // Imath headers.
 #ifdef APPLESEED_ENABLE_IMATH_INTEROP
 #include "foundation/platform/_beginexrheaders.h"
+#if __GNUC__ >= 7
```

Comment:
I am terribly sorry, but I forgot to mention that we have special headers specifically for this use case!

Please, have a look at the `#include "foundation/platform/_beginexrheaders.h"` at line 42 and `#include "foundation/platform/_endexrheaders.h"` at line 51. These headers work as "brackets" in which we place all OpenEXR includes.

If you put lines 43-46 in the above header and lines 48-50 in the header below, we can avoid further duplication these pragmas and the code will become cleaner.